### PR TITLE
Set an upper limit on future tasks

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

There was some discussion about what the upper limit should be. I went for a conservative option of 6 months but open to suggestion!

medic/cht-core#6564

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Review by dianabarsan — COMMENTED** [review]
Thanks a lot for working on this!  Nice work! 

I think 6 months is quite safe. 

I added one comment about time terminology (which is a special kind of confusing for me as a non-english speaker) and one request for test coverage.

File: shared-libs/rules-engine/src/task-states.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ const moment = require('moment');
  * recorded as docs.
  */
 const TIMELY_WHEN_NEWER_THAN_DAYS = 60;
+const TIMELY_WHEN_YOUNGER_THAN_DAYS = 180;
```
[the remark this answers is not in the document]
Comment (garethbowen) [reply shared-libs/rules-engine/src/task-states.js:21]: Yeah I'm not a fan of adding a negative. I've had a go at simplifying this... let me know what you think.

File: shared-libs/rules-engine/test/task-states.spec.js [hunk 2]
```diff
@@ -162,12 +162,32 @@ describe('task-states', () => {
...
+        readyStart: 0,
+        readyEnd: 0,
+      };
+      expect(TaskStates.isTimely(emission, Date.now())).to.be.false;
     });
 
     it('new emission is timely', () => {
       const emission = mockEmission(-MS_IN_DAY);
-      expect(TaskStates.isTimely(emission)).to.be.true;
+      expect(TaskStates.isTimely(emission, Date.now())).to.be.true;
```
[the remark this answers is not in the document]
Comment (garethbowen) [reply shared-libs/rules-engine/test/task-states.spec.js:214]: Added some tests for startDate and endDate, and threshold testing for date.

**Review by dianabarsan — APPROVED** [review]
Love it!

**Comment by latin-panda** [comment]
Should we leave a hint of the limit in the cht-docs?

**Comment by garethbowen** [comment]
@latin-panda Good call! PR created here: https://github.com/medic/cht-docs/pull/684

**Pull request merged**